AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -113,28 +112,35 @@ - (void)setParent:(id<ASLayoutable>)parent
   }
 }
 
-- (void)setChild:(id<ASLayoutable>)child;
+- (void)setChild:(id<ASLayoutable>)child
 {
   ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
-  
+  ASDisplayNodeAssertNotNil(child, @"Child cannot be nil");
+
   id<ASLayoutable> finalLayoutable = [self layoutableToAddFromLayoutable:child];
-  _child = finalLayoutable;
-  [self propagateUpLayoutable:finalLayoutable];
+  if (finalLayoutable) {
+    _children = @[finalLayoutable];
+    [self propagateUpLayoutable:finalLayoutable];
+  }
 }
 
 - (void)setChild:(id<ASLayoutable>)child forIdentifier:(NSString *)identifier
 {
   ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
+  ASDisplayNodeAssertNotNil(child, @"Child cannot be nil");
   
   id<ASLayoutable> finalLayoutable = [self layoutableToAddFromLayoutable:child];
   self.childrenWithIdentifier[identifier] = finalLayoutable;
+  if (finalLayoutable) {
+    self.children = [self.children arrayByAddingObject:finalLayoutable];
+  }
   
   // TODO: Should we propagate up the layoutable at it could happen that multiple children will propagated up their
   //       layout options and one child will overwrite values from another child
   // [self propagateUpLayoutable:finalLayoutable];
 }
 
-- (void)setChildren:(NSArray *)children
+- (void)setChildren:(NSArray<id<ASLayoutable>> *)children
 {
   ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
```

Comment:
@maicki and I talked about this. Currently ASLayoutSpec is wrapped with `NS_ASSUME_NONNULL_BEGIN` and we thought that it didn't make a lot of sense to send in nil children.

But interestingly enough, I noticed that ASBackgroundSpec has the background spec as nullable. Soooooo, not sure what we should do here. @maicki ?